sasCore/timelinethread.cpp

```diff
@@ -5,80 +5,166 @@
 #include <map>
 #include <list>
 #include <atomic>
+#include <assert.h>
 
 #include <include/sasCore/notifier.h>
 
+#include <iostream>
+
 namespace SAS {
 
     struct TimelineThread::Private
 	{
 		Notifier notif;
 
-		struct Entry
+        struct Entry : public TimelineThread::Entry
 		{
             using Ptr = std::shared_ptr<Entry>;
-			Id id = 0;
-			std::chrono::system_clock::time_point timestamp;
-            std::chrono::system_clock::duration period;
-            long remaining = 1;
+            Id id = 0;
+            std::chrono::system_clock::time_point m_timestamp;
+            std::chrono::system_clock::duration m_period;
+            long m_remaining = 1;
 			Func func;
+            std::string m_handlerDescriptor;
             bool done = false;
-            std::mutex mut;
+            std::recursive_mutex mut;
+            std::atomic_bool active;
+            std::function<void(const Entry::Ptr &)> onChanged;
+
+            void callOnChanged(const Entry::Ptr & e)
+            {
+                std::unique_lock<std::recursive_mutex> __locaker(mut);

```

Comment:
```suggestion
                std::unique_lock<std::recursive_mutex> __locaker(mut);

```
locaker, a régi jó barát :)